pyproject.toml

```diff
@@ -73,9 +73,7 @@ dependencies = [
     "tabulate>=0.9",
     "tqdm>=4.60",
     "uncertainties>=3.1.4",
-    # NumPy documentation suggests pinning the current major version as the C API is used
-    # https://numpy.org/devdocs/dev/depending_on_numpy.html#runtime-dependency-version-ranges
-    "numpy>=1.25.0,<3",
```

Comment:
I believe it should be [safe to remove the upper bound](https://numpy.org/devdocs/dev/depending_on_numpy.html#runtime-dependency-version-ranges) (especially considering numpy 3 wouldn't come anytime soon):
> Most libraries that rely on NumPy will not need to set an upper version bound: NumPy is careful to preserve backward-compatibility.

I think the original manual might have been misread here (ping @njzjz in case I'm wrong here):
> That said, if you are (a) a project that is guaranteed to release frequently, (b) use a large part of NumPy’s API surface, and (c) is worried that changes in NumPy may break your code, you can set an upper bound of <MAJOR.MINOR + N with N no less than 3, and MAJOR.MINOR being the current release of NumPy

If some project is very concerned (I assume not us) with numpy compatibility, they should pin `<MAJOR.MINOR + N` (not `<3`), which should be `<2.4.0` if current version is `2.1.0`.

Reply:
> If you use the NumPy C API (directly or via Cython), you can also pin **the current major version** to prevent ABI breakage.

![image](https://github.com/user-attachments/assets/86906607-984c-483f-bbfa-a38c5f89f70d)

Reply:
Cool! Thanks for the input. It's a bit unclear to me what "pin the current major version" refers to? I thought it refers to constrain the major numpy version to 2 (`2.0.0 <numpy<3`)?

But in any case, I don't think numpy 3 would be a concern anytime soon, what do you think?

Reply:
[NEP 53](https://numpy.org/neps/nep-0053-c-abi-evolution.html) has some explanations. In short, the ABI change in Numpy 2.0 is limited, and NumPy 2 ABI can be used with NumPy 1, but NumPy 1 ABI cannot be used with NumPy 2. However, there is no promise for the future NumPy version. For example, NumPy 3 ABI may or may not be able to be used with NumPy 2 - this is only known when NumPy 3 is developed.

Reply:
Yes I agree with that, but if I understand that manual correctly, a project should either:
- pin `<MAJOR.MINOR + N` (currently as `< 2.4`), or
- pin the current major version (`2 <= numpy < 3`)

Currently we are doing: `1.25 < numpy < 3`, if I understand correctly, "pin the current major version" refers to "setting the numpy major version as exactly 2" instead of "setting an upper bound for the major version"?